### DISCUSSION TITLE
Change "float" precision in Code

### DISCUSSION THREAD

**Discussion by Sorry777** [perspective-dev/perspective#1458] [Q&A] [answered]
Hello, how can i change "float" precision in Code, not on site? like here https://github.com/finos/perspective/pull/1386

**Comment by texodus** [comment, marked as answer]
There are two APIs for this available currently, both work but you likely want the latter (2) option if you're using a new enough Perspective to support it `0.7.0`+.

1) Use [`perspective config`](https://perspective.finos.org/docs/md/js.html#customizing-behavior-with-perspectiveconfigjs), like this

```javascript
const worker = window.perspective.worker({
    types: {
        float: {
            format: {
                style: "decimal",
                minimumFractionDigits: 6,
                maximumFractionDigits: 6
            }
        }
    }
});
```

2) Use `plugin_config` which is the serialized state of the user-controllable precision in #1386 .  Just configure the grid using the UI controls, then use the token returned by `save()` from the JS Console, copy this object and use it as an argument to `restore()`, as per the [this example in the docs](https://perspective.finos.org/docs/md/js.html#setting--reading-perspective-viewer-configuration-via-attributes).   

```javascript
token = document.querySelector("perspective-viewer#one").save(); // will be `async` in `0.10.0`

// ... later ...

document.querySelector("perspective-viewer#two").restore(token);
```

If you'd like, you can inspect this token and customize the`plugin_config` property

<img width="649" alt="Screen Shot 2021-06-23 at 3 41 37 PM" src="https://user-images.githubusercontent.com/60666/123158261-8c7a9400-d439-11eb-9322-74da2f1264ff.png">

**Reply by Sorry777** [reply]
Thanks
